### ISSUE TITLE
Three rules read the head of a valid `@charset`, and `at-rule-name-case` under `upper` leaves a file declaring no encoding

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#703]
### What minimal example or steps are needed to reproduce the bug?

`@charset` is not an at-rule a formatter is free to respell. The fallback-encoding step of [css-syntax-3](https://drafts.csswg.org/css-syntax/#input-byte-stream) reads it out of the byte stream before anything is parsed, and it matches the bytes `@charset "` exactly — one U+0020 and one double quote — then the label up to the next `"`, which has to be followed by `;`. Any other spelling declares no encoding at all, and the stylesheet falls back to the environment's.

```css
@charset "UTF-8";
a { color: pink; }
```

Measured over the whole registry, every rule under each primary `scripts/oracles/options.ts` names, with `fix: true`. Three rules speak of that head:

| Rule | Option | What it does today |
| --- | --- | --- |
| `at-rule-name-case` | `upper` | writes `@CHARSET "UTF-8";`, and the file declares no encoding any more |
| `at-rule-name-newline-after` | `always` | `1:8 Expected newline after at-rule name "@charset"`; no fixer |
| `at-rule-semicolon-space-before` | `always` | `1:16 Expected single space before ";"`; no fixer |

Only the first one writes. The second gets a fixer in #696, and that fixer would write `@charset\n"UTF-8";`.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/at-rule-name-case": "upper"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

An encoding declaration spelled the way the specification reads it is left alone, as `string-quotes` already leaves it alone — it writes the double quotes `@charset` takes and nothing else.

### What actually happened?

`--fix` returns `@CHARSET "UTF-8";` and the next run is silent, the rule having got what it asked for. The stylesheet has lost the declaration a decoder reads, and nothing reports it.

### Do you have a proposal to fix the bug?

One reading, shared by the three: an at-rule is the file's encoding declaration where it stands first in its root and its head is spelled as the algorithm reads it. Each of them passes over such a node. Every other spelling is untouched by the reading, since it declares no encoding for a fix to lose.

Whether the invalid spellings are reported at all is not this plugin's question — that is [stylelint/stylelint#7492](https://github.com/stylelint/stylelint/issues/7492), which asks Stylelint for an `at-charset-rule-no-invalid` rule of its own. This issue is only about the plugin not rewriting the valid one.

`unicode-bom` under `always` reports `1:1` here for a reason of its own, and a byte-order mark in front of a `@charset` naming anything but UTF-8 does override it. That rule carries no fixer, so nothing is written; noted rather than filed.

Found while writing the fixer of #696.
